src/llvm-multiversioning.cpp

```diff
@@ -697,18 +698,59 @@ Constant *CloneCtx::rewrite_gv_init(const Stack& stack)
     return res;
 }
 
+// replace an alias to a function with a trampoline and (uninitialized) global variable slot
+std::pair<Function *, GlobalVariable *> CloneCtx::rewrite_alias(GlobalAlias *alias)
+{
+    Function *F = cast<Function>(alias->getAliasee());
+
+    auto Name = alias->getName();
+    alias->setName("");
+    Function *trampoline =
+        Function::Create(F->getFunctionType(), alias->getLinkage(), Name, &M);
+    trampoline->copyAttributesFrom(F);
+    alias->eraseFromParent();
+    GlobalVariable *slot =
```

Comment:
I believe this can use the cached version `get_reloc_slot`.

Reply:
I originally did so, but those slots are of type `void*` whereas the initializer set by `fix_gv_uses` / `rewrite_gv_init` has the original function type.

Reply:
The type shouldn't matter at all. The one for gv has the type only because it is not used by anyone else and the slot is already there so there isn't any better option.

Reply:
LLVM doesn't like it though:
```
llvm-9.0.1/lib/IR/Globals.cpp:394: void llvm::GlobalVariable::setInitializer(llvm::Constant*): Assertion `InitVal->getType() == getValueType() && "Initializer type must match GlobalVariable type"' failed.
```

Reply:
It shouldn't need any initialization.